src/main/java/uk/gov/companieshouse/psc/extensions/api/service/impl/PscExtensionsServiceImpl.java

```diff
@@ -35,8 +39,37 @@ public PscExtension save(PscExtension filing) {
      * @param filingId the Filing ID
      * @return the stored entity if found
      */
+
     @Override
     public Optional<PscExtension> get(String filingId) {
         return repository.findById(filingId);
     }
+
+    /**
+     * Query the mongoDB for the number psc extension requests.
+     *
+     * @param pscNotificationId   the PSC ID
+     * @return the number of psc extension requests if found.
+     *
+     */
+
+    @Override
+    public Optional<Long> getExtensionCount(String pscNotificationId) {
+        if (pscNotificationId == null || pscNotificationId.isEmpty()) {
+            logger.error("Provided notification ID is missing");
+            throw new NullPointerException("Notification ID cannot be null or empty");
+        }
+
+        long count = repository.countByDataPscNotificationId(pscNotificationId);
+
+        if (count > 1) {
+            logger.error("Number extension requests exceeded for " + pscNotificationId);
+            throw new IllegalArgumentException();
+
```

Comment:
Is such validation required? If so, should it be here? What response will the REST request client get if this happens?